qiskit_nature/properties/second_quantization/electronic/integrals/two_body_electronic_integrals.py

```diff
@@ -164,7 +164,7 @@ def to_spin(self) -> np.ndarray:
         if self._basis == ElectronicBasis.SO:
             return self._matrices  # type: ignore
 
-        so_matrix = np.zeros([2 * s for s in self._matrices[0].shape])
+        so_matrix = np.zeros([2 * s for s in self._matrices[0].shape],dtype = 'complex')
```

Comment:
This change to complex will end up consuming a bunch more memory (double) since commonly we have real values for these integrals I think. I am also trying to figure out how this change factors in since the UCC operators are built directly as FermionicOps and not via integrals right.